src/ZkFold/Base/Protocol/ARK/Plonk.hs

```diff
@@ -44,11 +47,12 @@ type G2 = Point BLS12_381_G2
 data Plonk (d :: Natural) (n :: Natural) t = Plonk F F F (Vector n Natural) (ArithmeticCircuit 1 F) F
     deriving (Show)
 -- TODO (Issue #25): make a proper implementation of Arbitrary
-instance Arbitrary (Plonk d n t) where
+instance (KnownNat d) => Arbitrary (Plonk d n t) where
     arbitrary = do
-        let (omega, k1, k2) = getParams 5
+        let nP = ceiling @Double . logBase 2.0 . fromIntegral $ value @d
+        let (omega, k1, k2) = getParams nP
         ac <- arbitrary
-        Plonk omega k1 k2 (Vector [1, 2]) ac <$> arbitrary
+        Plonk omega k1 k2 (Vector [1..nP]) ac <$> arbitrary
```

Comment:
`nP` is not related to the number of inputs. The 4th field of `Plonk` type is the vector of indices of public inputs. It should be a subset of circuit inputs.